src/main/java/com/minecolonies/coremod/colony/ColonyManager.java

```diff
@@ -992,49 +1008,128 @@ public static void onWorldLoad(@NotNull final World world)
         }
     }
 
-    public static boolean backupColonyData()
+    /**
+     * Initiates a timer to schedule the colony backup task
+     * backupDelay of 0 disables automatic backup
+     */
+    public static void scheduleColonyBackup()
     {
-        if (numWorldsLoaded > 0 && saveNeeded)
+        final int backupDelay = Configurations.administration.colonyBackupTimeMins;
+        if (backupDelay > 0)
         {
-            saveColonies(false);
+            backupTimerTask = new backupColonyTimerTask();
+            backupTimer.schedule(backupTimerTask, backupDelay * 60 * 1000);
         }
+    }
+
+    /**
+     * Task to be called from backupTimer, triggers main backup method
+     * by overriding run() method of base TimerTask
+     */
 
-        try(FileOutputStream fos = new FileOutputStream(getBackupSaveLocation(new Date())))
+    private static class backupColonyTimerTask extends TimerTask
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this class name to match the regular expression '^[A-Z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00101)